packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -515,72 +520,88 @@ class HeapTreeViewState extends State<HeapTree>
-                    controller.treeChanged();
+      ),
+      const SizedBox(width: defaultSpacing),
+      _groupByDropdown(textTheme),
+      const SizedBox(width: defaultSpacing),
+      // TODO(terry): Mechanism to handle expand/collapse on both tables
+      // objects/fields. Maybe notion in table?
+      FixedHeightOutlinedButton(
+        buttonKey: collapseAllButtonKey,
+        tooltip: 'Collapse All',
+        onPressed: snapshotDisplay is MemoryHeapTable
+            ? () {
+                MemoryScreen.gaAction(key: collapseAllButtonKey);
+                if (snapshotDisplay is MemoryHeapTable) {
+                  controller.groupByTreeTable.dataRoots.every((element) {
+                    element.collapseCascading();
+                    return true;
+                  });
+                  if (controller.instanceFieldsTreeTable != null) {
+                    // We're collapsing close the fields table.
+                    controller.selectedLeaf = null;
                   }
+                  // All nodes collapsed - signal tree state changed.
+                  controller.treeChanged();
                 }
-              : null,
-          child: const Icon(
-            Icons.vertical_align_top,
-            size: defaultIconSize,
-          ),
+              }
+            : null,
+        child: const Icon(
+          Icons.vertical_align_top,
+          size: defaultIconSize,
         ),
-        FixedHeightOutlinedButton(
-          buttonKey: expandAllButtonKey,
-          tooltip: 'Expand All',
-          onPressed: snapshotDisplay is MemoryHeapTable
-              ? () {
-                  MemoryScreen.gaAction(key: expandAllButtonKey);
-                  if (snapshotDisplay is MemoryHeapTable) {
-                    controller.groupByTreeTable.dataRoots.every((element) {
-                      element.expandCascading();
-                      return true;
-                    });
-                  }
-                  // All nodes expanded - signal tree state  changed.
-                  controller.treeChanged();
+      ),
+      FixedHeightOutlinedButton(
+        buttonKey: expandAllButtonKey,
+        tooltip: 'Expand All',
+        onPressed: snapshotDisplay is MemoryHeapTable
+            ? () {
+                MemoryScreen.gaAction(key: expandAllButtonKey);
+                if (snapshotDisplay is MemoryHeapTable) {
+                  controller.groupByTreeTable.dataRoots.every((element) {
+                    element.expandCascading();
+                    return true;
+                  });
+                }
+                // All nodes expanded - signal tree state  changed.
+                controller.treeChanged();
+              }
+            : null,
+        child: const Icon(
+          Icons.vertical_align_bottom,
+          size: defaultIconSize,
+        ),
+      ),
+      const SizedBox(width: defaultSpacing),
+      Container(
+        width: 170,
+        child: SwitchListTile(
```

Comment:
Just use a Row with text and a switch as children instead of the SwitchListTile. Also, remove the wrapping container. If you give the Row MainAxisSize.min, that should be the behavior you want

Reply:
Done.